testsuite/integration/src/test/java/org/keycloak/testsuite/RealmResourceTest.java

```diff
@@ -0,0 +1,67 @@
+package org.keycloak.testsuite;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.services.managers.RealmManager;
+import org.keycloak.testsuite.rule.KeycloakRule;
+import org.keycloak.testsuite.rule.KeycloakRule.KeycloakSetup;
+import org.keycloak.testsuite.rule.WebResource;
+import org.keycloak.testsuite.rule.WebRule;
+import org.openqa.selenium.WebDriver;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author <a href="mailto:thomas.darimont@gmail.com">Thomas Darimont</a>
+ */
+public class RealmResourceTest {
+
+    @ClassRule
+    public static KeycloakRule keycloakRule = new KeycloakRule(new KeycloakSetup() {
+
+        @Override
+        public void config(RealmManager manager, RealmModel adminstrationRealm, RealmModel appRealm) {
+
+            RealmModel testRealm = manager.getRealmByName("test");
+
+            ClientModel launchpadClient = testRealm.addClient("launchpad-test");
+            launchpadClient.setBaseUrl("");
+            launchpadClient.setRootUrl("http://example.org/launchpad");
+
+            ClientModel dummyClient = testRealm.addClient("dummy-test");
+            dummyClient.setRootUrl("http://example.org/dummy");
+            dummyClient.setBaseUrl("/base-path");
+        }
+    });
+
+    @Rule
+    public WebRule webRule = new WebRule(this);
+
+    @WebResource
+    protected OAuthClient oauth;
+
+    @WebResource
+    protected WebDriver webDriver;
+
+    /**
+     * Integration test for {@link org.keycloak.services.resources.RealmsResource#getRedirect(String, String)}.
+     * @throws Exception
+     */
+    @Test
+    public void testClientRedirectEndpoint() throws Exception {
+
+        oauth.doLogin("test-user@localhost", "password");
+
+        webDriver.get("http://localhost:8081/auth/realms/test/clients/launchpad-test/redirect");
```

Comment:
Thanks I made some progress with this - will push the new test cases soon, though the initial tests were much easier to write...